# Home Work 5

In [1]:
import numpy as np
from scipy import stats

## Задача 1

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания $\mu$ с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

### Решение:

Задано $\sigma = 16$, $p = 0.95$, $\overline{X} = 80$, $n = 256$. Так как генеральная совокупность распределена нормально, то доверительный интервал оценки матожидания $\mu$ можно найти по формуле

$$
P \left(
    \overline{X} - t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
    \leq \mu \leq 
    \overline{X} + t_{1 - \alpha / 2} \cdot \dfrac{\sigma}{\sqrt{n}}
\right) = p.
$$
Здесь $\alpha = 1 - p = 0.05$. Так как мы работаем со случайной величиной, а не с выборкой, то квантиль $t_{1 - \alpha / 2}$ найдём из стандартного нормального распределения.

In [2]:
alpha = 0.05
sigm = 16
p = 0.95
mean = 80
n = 256

alpha = 1 - p
t = stats.norm.ppf(1 - alpha/2)

low = mean - t * (sigm / np.sqrt(n))
up = mean + t * (sigm / np.sqrt(n))
print(f'Low = {round(low, 2)}, Up = {round(up, 2)}')

Low = 78.04, Up = 81.96


### Ответ:

Доверительный интервал для оценки матожидания с надёжностью 0.95:

$78.04 \leq \mu \leq 81.96$.

## Задача 2

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

202, 203, 199, 197, 195, 201, 200, 204, 194, 190

Известно, что их веса распределены нормально.
1. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?
2. Найдите P-значение для данного теста.

### Решение:

1. Проверим гипотезу $H_0$ о том что матожидание веса пачки печенья равно 200г. Так как веса пачек печенья распределены нормально, то используем статистику: $$T(X) = \dfrac{\overline{X} - a}{\sigma_X / \sqrt{n}},$$ 

здесь $a = 200$, $n = 10$.

In [3]:
X = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
a = 200
n = len(X)
alpha = 0.01
def statistic(samples):
    return (samples.mean() - a) / (samples.std(ddof=1) / np.sqrt(samples.shape[0]))

Статистика $T(X)$ будет иметь распределение Стьюдента. Критическая область: $$\Omega_\alpha = (-\infty, t_{\alpha/2, \: n-1}) \cup (t_{1 - \alpha/2, \: n-1}, \infty).$$ Найдём квантили $t_{\alpha/2, \: n - 1}$ и $t_{1 - \alpha/2, \: n - 1}$.

In [4]:
t1 = stats.t.ppf(alpha/2, df=n-1)
t2 = stats.t.ppf(1 - alpha/2, n-1)
t1, t2

(-3.24983554401537, 3.2498355440153697)

Получили критическую область $\Omega_\alpha = (-\infty, -3.25) \cup (3.25, \infty)$. 

Посчитаем $T(X)$ для нашей выборки.

In [5]:
T = statistic(X)
T

-1.0651074037450896

Получили $T(X) = -1.065$. Эта точка не попадает в критическую область, поэтому гипотеза $H_0$ остаётся в силе. То есть утверждение продавца истинно с уровнем значимости 1%.

2. Найдём P-значение для данного теста.

In [6]:
p_left = stats.t.cdf(T, df=n - 1)
p_right = 1 - stats.t.cdf(T, df=n - 1)

pvalue = 2 * min(p_left, p_right)

pvalue

0.31456660352472854

### Ответ:

1. Оснований отклонить гипотезу нет.
2. P-значение для данного теста: pvalue = 0.315

## Задача 3

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

### Решение:

Проверяется гипотеза $H_0$ о том что матожидание распределения диаметра шариков не превышает 17мм. Так как распределение нормальное и с известной дисперсией, то для проверки гипотезы $H_0$ будем использовать статистику $T(X) = \dfrac{\overline{X} - a}{\sigma / \sqrt{n}}$. Эта статистика имеет стандартное нормальное распределение. Для нашей выборки оказалось что $\overline{X}=17.5$. $n = 100$, $\sigma = 2мм$, $a = 17мм$. То есть$$T(X) = \dfrac{\overline{X} - 17}{2 / \sqrt{100}} = (\overline{X} - 17) \cdot 5$$

При уровне значимости $\alpha = 0.025$ правосторонняя критическая область $\Omega_\alpha = (t_{1 - \alpha}, \infty)$. Посчитаем квантиль $t_{1 - \alpha}$.

In [7]:
alpha = 0.025
t2 = stats.norm.ppf(1 - alpha)
t2

1.959963984540054


Получили правостороннюю критическую область $\Omega_\alpha = (1.96, \infty)$. Посчитаем значение статистики $T(X)$ для $\overline{X} = 17.5$.

In [8]:
T = lambda x: (x - 17) * 5
T(17.5)

2.5

### Ответ:

Для нашей конкретной выборки статистика $T(17.5) = 2.5$ попадает в критическую область, поэтому гипотеза $H_0$ отвергается. То есть матожидание распределения шариков превышает 17мм.